test/services/concerns/xccdf/test_result_test.rb

```diff
@@ -38,4 +38,46 @@ class Mock
       @mock.save_test_result
     end
   end
+
+  test 'old test results within a policy replaced by the new test result' do
+    profiles(:two).update!(policy_object: policies(:one),
+                           account: accounts(:test))
+    profiles(:one).update!(policy_object: policies(:one),
+                           account: accounts(:test))
+    TestResult.where(host: hosts(:one), profile: profiles(:one)).destroy_all
+
+    assert_difference('TestResult.count' => 2) do
+      TestResult.create!(host: hosts(:one),
+                         profile: profiles(:one),
+                         end_time: @end_time - 3.minutes)
+
+      TestResult.create!(host: hosts(:one),
+                         profile: profiles(:two),
+                         end_time: @end_time - 8.minutes)
+    end
+
+    assert_difference('TestResult.count' => -1) do
+      @mock.save_test_result
```

Comment:
Can you please also check which test result is kept?

Reply:
Updated to check the remaining test result is the correct one.